src/AcceptanceTests/Routing/When_unsubscribing_from_one_of_the_events_for_ForwardingTopology.cs

```diff
@@ -0,0 +1,138 @@
+﻿namespace NServiceBus.Azure.Transports.WindowsAzureServiceBus.AcceptanceTests.Routing
+{
+    using System;
+    using System.Linq;
+    using System.Security.Cryptography;
+    using System.Text;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using Features;
+    using Microsoft.ServiceBus;
+    using NUnit.Framework;
+    using Settings;
+    using Transport.AzureServiceBus;
+    using AzureServiceBus;
+    using NServiceBus.AcceptanceTests;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+
+    public class When_unsubscribing_from_one_of_the_events_for_ForwardingTopology : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_remove_subscription_rule_from_topology()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(b => b.When((session, c) => session.Unsubscribe<MyEvent>()))
+                .Done(c => c.EndpointsStarted)
+                .Run();
+
+            var connectionString = Environment.GetEnvironmentVariable("AzureServiceBusTransport.ConnectionString");
+            var namespaceManager = NamespaceManager.CreateFromConnectionString(connectionString);
+            var rawEndpointName = ConfigureEndpointAzureServiceBusTransport.NameForEndpoint<Endpoint>();
+            var endpointName = MD5HashBuilder.Build(rawEndpointName);
+            var ruleName = MD5HashBuilder.Build(typeof(MyEvent).FullName);
+
+            if (context.IsForwardingTopology)
+            {
+                var isSubscriptionFound = await namespaceManager.SubscriptionExistsAsync("bundle-1", endpointName);
+                Assert.IsTrue(isSubscriptionFound, "Subscription under 'bundle-1' should have been found, but it wasn't.");
+
+                isSubscriptionFound = await namespaceManager.SubscriptionExistsAsync("bundle-2", endpointName);
+                Assert.IsTrue(isSubscriptionFound, "Subscription under 'bundle-2' should have been found, but it wasn't.");
+
+                var rules = await namespaceManager.GetRulesAsync("bundle-1", endpointName);
+                Assert.That(rules.Any(rule => rule.Name == ruleName), Is.False);
```

Comment:
I.e. for this scenario it is known that there will be _other subscriptions_, one or many shouldn't matter. What should matter is that 
1. The subscription for the event we've unsubscribed from is not found
1. Subscription entity remains (due to other event subscriptions still remaining)

Since there are only two events subscribed to, one of them no longer exists, but subscription entity is there, backed up with the 2nd test (`when_unsubscribing`) it defacto asserts that the second event subscription was not removed.